src/dsp_tools/commands/xmlupload/models/ontology_lookup_models.py

```diff
@@ -50,28 +52,109 @@ def _extract_classes_and_properties_from_onto(onto_json: list[dict[str, Any]]) -
     Returns:
         Ontology instance with the classes and properties
     """
-    classes = _get_all_cleaned_classes_from_json(onto_json)
-    properties = _get_all_cleaned_properties_from_json(onto_json)
+    classes = _get_all_cleaned_classes_from_onto(onto_json)
+    properties = _get_all_cleaned_properties_from_onto(onto_json)
     return OntoInfo(classes, properties)
 
 
-def _get_all_cleaned_classes_from_json(onto_json: list[dict[str, Any]]) -> list[str]:
-    classes = _get_all_classes_from_json(onto_json)
+def _get_all_cleaned_classes_from_onto(onto_json: list[dict[str, Any]]) -> list[str]:
+    classes = _get_all_classes_from_onto(onto_json)
     return _remove_prefixes(classes)
 
 
-def _get_all_classes_from_json(onto_json: list[dict[str, Any]]) -> list[str]:
+def _get_all_classes_from_onto(onto_json: list[dict[str, Any]]) -> list[str]:
     return [elem["@id"] for elem in onto_json if elem.get("knora-api:isResourceClass")]
 
 
-def _get_all_cleaned_properties_from_json(onto_json: list[dict[str, Any]]) -> list[str]:
-    props = _get_all_properties_from_json(onto_json)
+def _get_all_cleaned_properties_from_onto(onto_json: list[dict[str, Any]]) -> list[str]:
+    props = _get_all_properties_from_onto(onto_json)
     return _remove_prefixes(props)
 
 
-def _get_all_properties_from_json(onto_json: list[dict[str, Any]]) -> list[str]:
+def _get_all_properties_from_onto(onto_json: list[dict[str, Any]]) -> list[str]:
     return [elem["@id"] for elem in onto_json if not elem.get("knora-api:isResourceClass")]
 
 
 def _remove_prefixes(ontology_elements: list[str]) -> list[str]:
     return [x.split(":")[1] for x in ontology_elements]
+
+
+AllowedEncodings = Literal["utf8", "xml"]
+
+
+@dataclass
+class TextValueData:
+    resource_id: str
+    property_name: str
+    encoding: AllowedEncodings
+
+
+@dataclass
+class PropertyTextValueEncodingTypes:
+    """
+    This class contains the information
+    which properties have which type of encoding for a TextValue property in the ontology.
+    """
+
+    formatted_text_props: set[str] = field(default_factory=set)
+    unformatted_text_props: set[str] = field(default_factory=set)
+
+
+def get_text_value_properties_and_formatting_from_onto(
```

Comment:
I don't think using encoding is the better choice. Because the type in fact has nothing to do with the encoding, it is just how we called it in the XML, and in the ontology it is referred to differently. But I changed it so that it is more conform with the naming in the ontology, as this deals with that and not the XML.

Reply:
👍